tools/spatialVideo/index.js

```diff
@@ -0,0 +1,180 @@
+/* global Envelope, SpatialInterface */
+import { VideoUI, VideoUIStates } from './scripts/VideoUI.js';
+
+const MAX_RECORDING_DURATION = 20000;
+
+let videoUI;
+let videoPlayback;
+let recordingActive = false;
+let virtualizerRecordingTimeout = null;
+let defaultUrls;
+
+// The following timeout and interval allow for one instance of the tool to declare itself the leader and be in charge of synchronizing state
+// Heartbeat-style status updates are necessary to allow for new connections to know the current state rather than an outdated one
+let selfNominateTimeout;
+const selfNominateTimeoutDuration = 5000 + Math.random() * 1000; // Seconds before self-nomination
+let leaderBroadcastInterval;
+let leaderBroadcastIntervalDuration = 1000;
+let leaderId = 0;
```

Comment:
This is pretty cool! Not for this PR, but do you think having a "leader" is a common pattern that tool developers may need to implement and could be abstracted into some sort of APIs?